### PR TITLE
Task #179501 feat: Impersonate Login API

### PR DESCRIPTION
**Pull request by praneettekdi**

### CODE REVIEW

File: src/users/impersonatelogin.php [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+	/**
+	 * Generate key method
+	 *
+	 * @return  object|boolean
+	 */
+	public function keygen()
+	{
+		// Init variables
+		$obj              = new stdclass;
+		$jinput           = JFactory::getApplication()->input;
```
Comment (manojLondhe) [inline src/users/impersonatelogin.php]: JFactory => use non J prefix classses wherever available

File: src/users/impersonatelogin.php [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+		else
+		{
+			ApiError::raiseError("403", JText::_('PLG_API_USERS_BAD_REQUEST_MESSAGE'));
+
+			return false;
+		}
+
+		if ($userId == null)
+		{
+			$model  = FD::model('Users');
```
Comment (manojLondhe) [inline src/users/impersonatelogin.php:101]: FD is older library of Easysocial.

WIll this work without ES when email is passed?

Comment (praneettekdi) [reply src/users/impersonatelogin.php:101]: No...I have added another function to fetch email

File: src/users/impersonatelogin.php [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+	 *
+	 * @param   string   $email  User email
+	 *
+	 * @return  integer   User Id.
+	 *
+	 * @since   1.0
+	 */
+	private function getUserByEmail($email)
+	{
+		$db = JFactory::getDbo();
```
Comment (manojLondhe) [inline src/users/impersonatelogin.php]: JFactory > Factory

**Pull request merged**